### PR TITLE
17795: FAA Split Brain Data Model

### PR DESCRIPTION
**Pull request by saikumar9**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [17795](https://devops.dchbx.org/redmine/issues/17795) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @DFos85 |

### CODE REVIEW

File: app/models/financial_assistance/application.rb [hunk 1]
```diff
@@ -677,7 +689,7 @@ def unset_submit
 
   def create_tax_households
     ## Remove  when copy method is fixed to exclude copying Tax Household
-    tax_households.destroy_all
+    # tax_households.destroy_all #We don't need to do it anymore as TaxHouseholds are embeded in Household
```
Comment (billy-ran-away) [inline app/models/financial_assistance/application.rb]: This line should really just be removed.

File: app/models/household.rb [hunk 2]
```diff
@@ -329,13 +329,13 @@ def hbx_enrollments_with_aptc_by_year(year)
...
-  #   applications.where(assistance_year: year).each do |app|
-  #     app.eligibility_determinations.each do |ed|
-  #       eds << ed
-  #     end
-  #   end
-  #   eds
-  # end
+  def eligibility_determinations_for_year(year)
+    eds = []
+    tax_households.tax_household_with_year(year).each do |th|
```
Comment (billy-ran-away) [inline app/models/household.rb]: Do an `.inject` on the other loop starting an empty array, `[]`, and then inside add a tax household's `eligibility_determinations` to `.inject`'s `memo`.

File: app/models/tax_household.rb [hunk 3]
```diff
@@ -162,43 +173,36 @@ def size
...
+    household.family
   end
 
   #primary applicant is the tax household member who is the subscriber
   def primary_applicant
-    applicants.each do |applicant|
-      return applicant if applicant.family_member.is_primary_applicant?
+    #Review split brain
+    if application_id.present?
+      applicants.each do |applicant|
```
Comment (billy-ran-away) [inline app/models/tax_household.rb]: Shouldn't this need to be `application.applicants`? You deleted the `applicants` method itself below.

File: app/controllers/concerns/aptc.rb [hunk 4]
```diff
@@ -1,7 +1,9 @@
 module Aptc
   def get_shopping_tax_household_from_person(person, year)
     if person.present? && person.has_active_consumer_role?
-      person.primary_family.active_approved_application.latest_active_tax_households_with_year(year).first rescue nil
+      family = person.primary_family
+      person.primary_family.active_approved_application.latest_active_tax_households_with_year(year).first rescue nil if family.active_approved_application.present?
```
Comment (saikumar9) [inline app/controllers/concerns/aptc.rb]: conditional check: either - or

Comment (psk86) [inline app/controllers/concerns/aptc.rb]: conditional check: either - or

**Comment by saikumar9** [comment]
Old FAA code.

**Pull request closed without merging**